### PR TITLE
adding opentelemetry extension stubs

### PR DESCRIPTION
**Pull request by brettmc**
adding stubs for the opentelemetry extension: https://github.com/open-telemetry/opentelemetry-php-instrumentation
https://pecl.php.net/package/opentelemetry

### CODE REVIEW

**Comment by isfedorov** [comment]
@brettmc Does the extension really contain only one function? 
Could you please add folder name to the PhpCoreStubsProvider array to `tests/TestData/Providers/Stubs/PhpCoreStubsProvider.php:163`?

**Comment by brettmc** [comment]
> Does the extension really contain only one function?

Yes. Simple but powerful :)

**Comment by VencaKrecl** [comment]
I've just started work on OTel so the MR would help a lot :-D When will this MR be merged?

**Comment by isfedorov** [comment]
@VencaKrecl when the asked question above will be answered :)

**Comment by VencaKrecl** [comment]
@isfedorov Do you mean this answer? https://github.com/JetBrains/phpstorm-stubs/pull/1542#issuecomment-1687166056

**Comment by isfedorov** [comment]
> @isfedorov Do you mean this answer? [#1542 (comment)](https://github.com/JetBrains/phpstorm-stubs/pull/1542#issuecomment-1687166056)

🤦 to me. I've added comment to review but forgot to click "Submit review". Submitted now. The questions is https://github.com/JetBrains/phpstorm-stubs/pull/1542#pullrequestreview-1587535580

File: opentelemetry/opentelemetry.php [hunk 1]
```diff
@@ -0,0 +1,23 @@
+<?php
+
+namespace OpenTelemetry\Instrumentation;
+
+use Closure;
+
+/**
+ * @param string|null $class The (optional) hooked function's class. Null for a global/built-in function.
```
[the remark this answers is not in the document]
Comment (VencaKrecl) [reply opentelemetry/opentelemetry.php:8]: Yes, It is. You can add a hook for class function `hook('SplFileObject::class', 'fwrite')` or only for function `hook(null,'file_get_contents')`, there is official stub file https://github.com/open-telemetry/opentelemetry-php-instrumentation/blob/main/ext/opentelemetry.stub.php

[the remark this answers is not in the document]
Comment (isfedorov) [reply opentelemetry/opentelemetry.php:8]: @VencaKrecl passing `null` !== optional, optional would mean call `hook('file_get_contents') without passing any value as the first parameter

[the remark this answers is not in the document]
Comment (VencaKrecl) [reply opentelemetry/opentelemetry.php:8]: I tried `hook('file_get_contents')` here is the result `NOTICE: PHP message: PHP Fatal error:  Uncaught ArgumentCountError: OpenTelemetry\Instrumentation\hook() expects at least 2 arguments`. The first parameter is mandatory.

[the remark this answers is not in the document]
Comment (brettmc) [reply opentelemetry/opentelemetry.php]: thanks for the review, I've updated to use `string|null`

**Pull request merged**